www/js/lib/uiUtil.js

```diff
@@ -183,15 +183,27 @@ function slideAway (e) {
  * @param {String} closeMessageLabel  The text to display on the close alert message button (optional, Default = "Okay")
  * @param {String} hideOptionLabel  The text to display on the hide option button (optional, Default = "Don't ask again")
  * @param {Boolean} displayHideOption If true, option to permanently hide the modal will be shown (currently only implemented for hideExternalLinkWarning)
+ * @param {Object} archiveMetadata If warning message for unknown ZIM file, contains the information to show user to help them confirm validity of ZIM file
+ * @param {String} archiveMetadata.name The name of the ZIM file
+ * @param {String} archiveMetadata.creator The creator of the ZIM file
+ * @param {String} archiveMetadata.publisher The publisher of the ZIM file
+ * @param {String} archiveMetadata.scraper The scraper used to create the ZIM file
```

Comment:
Hmm. It's clunky to create these custom properties extending the systemAlert function for a single purpose in the app. The existing function works absolutely fine if you simply pass it the HTML with the text already translated. As this is a basic and generic function, I don't think you should alter it for such a specific purpose.